mypy/checker.py

```diff
@@ -3341,12 +3341,24 @@ def check_incompatible_property_override(self, e: Decorator) -> None:
                     self.fail(message_registry.READ_ONLY_PROPERTY_OVERRIDES_READ_WRITE, e)
 
     def visit_with_stmt(self, s: WithStmt) -> None:
+        exceptions_maybe_suppressed = False
         for expr, target in zip(s.expr, s.target):
             if s.is_async:
-                self.check_async_with_item(expr, target, s.unanalyzed_type is None)
+                exit_ret_type = self.check_async_with_item(expr, target, s.unanalyzed_type is None)
             else:
-                self.check_with_item(expr, target, s.unanalyzed_type is None)
-        self.accept(s.body)
+                exit_ret_type = self.check_with_item(expr, target, s.unanalyzed_type is None)
+            if is_literal_type(exit_ret_type, "builtins.bool", False):
```

Comment:
Are these rules/conventions documented somewhere? I think they should be, otherwise we will just forget them (either in mypy docs, btw do we have a section about context managers?, or in typeshed). Maybe also add a link here.

Reply:
I did add some notes about these conventions to the typeshed contribution page (at the bottom of the [conventions][0] section), but this conventions currently aren't otherwise documented.

We do mention context managers once on the [Protocols page](https://mypy.readthedocs.io/en/stable/protocols.html#context-manager-protocols) -- I'll add a note about this there in a follow-up PR.

  [0]: https://github.com/python/typeshed/blob/master/CONTRIBUTING.md#conventions